### PR TITLE
New module - iam_role

### PR DESCRIPTION
**Pull request by wimnat**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

iam_role
##### ANSIBLE VERSION

2.2
##### SUMMARY

Module to allow managing AWS IAM roles.  Similar functionality to current iam module but it was decided that the module should be split in to separate ones i.e. iam_role, iam_user, iam_group.

This module is the first of that split.

Extra functionality includes support for managed policies

### CODE REVIEW

**Comment by wimnat** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/2201

File: lib/ansible/modules/cloud/amazon/iam_role.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: iam_role
+short_description: Manage AWS IAM roles
+description:
+  - Manage AWS IAM roles
+version_added: "2.2"
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/iam_role.py]: 2.3

File: lib/ansible/modules/cloud/amazon/iam_role.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    connection = boto3_conn(module, conn_type='client', resource='iam', region=region, endpoint=ec2_url, **aws_connect_params)
+
+    state = module.params.get("state")
+
+    if state == 'present':
+        create_or_update_role(connection, module)
+    else:
+        destroy_role(connection, module)
+
+from ansible.module_utils.basic import *
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/iam_role.py]: Please import only what you need, rather than `*`.

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

**Comment by gundalow** [comment]
Given the reviews in https://github.com/ansible/ansible-modules-extras/pull/2201 once my comments have been addressed I think we can merge this

File: lib/ansible/modules/cloud/amazon/iam_role.py (hunk 2 above)
Comment (wimnat) [reply lib/ansible/modules/cloud/amazon/iam_role.py]: Is this a new requirement?  I've always written my modules like this.  Is there doc for likely necessary includes?

Comment (gundalow) [reply lib/ansible/modules/cloud/amazon/iam_role.py]: http://docs.ansible.com/ansible/dev_guide/developing_modules.html#common-module-boilerplate

**Comment by mattclay** [comment]
@wimnat The file `test/integration/targets/win_file/files/foobar/directory fileD` is unrelated to this PR and appears to have been added accidentally.

**Comment by wimnat** [comment]
@gundalow all done - just the specific import now
@mattclay - file removed

**Review by gundalow — APPROVED** [review]

**Pull request merged**